tensorflow/core/framework/device_factory.h

```diff
@@ -30,8 +30,9 @@ struct SessionOptions;
 class DeviceFactory {
  public:
   virtual ~DeviceFactory() {}
-  static void Register(const std::string& device_type, DeviceFactory* factory,
-                       int priority);
+  static void Register(const std::string& device_type,
+                       const std::string& subdevice_type,
+                       DeviceFactory* factory, int priority);
```

Comment:
I assume this will be called by the plugin?  Will the plugin use a higher priority to overwrite the `subdevice_type` for the built-in `GPU` device?

Reply:
This is called by TensorFlow proper. As discussed in the [RFC](https://github.com/tensorflow/community/blob/master/rfcs/20200624-pluggable-device-for-tensorflow.md), proper will set a higher priority for the plugged device and will overwrite the built-in GPU device if the plugin registers a GPU device type.